mongodb/trigger/eventlistener/trigger.go

```diff
@@ -0,0 +1,266 @@
+package eventlistener
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	"github.com/project-flogo/core/trigger"
+	mongodb "github.com/project-flogo/datastore-contrib/mongodb/connector/connection"
+	"go.mongodb.org/mongo-driver/bson"
+	mongo "go.mongodb.org/mongo-driver/mongo"
+)
+
+var triggerMd = trigger.NewMetadata(&Settings{}, &HandlerSettings{}, &Output{})
+
+func init() {
+	_ = trigger.Register(&Trigger{}, &TriggerFactory{})
+}
+
+// TriggerFactory My Trigger factory
+type TriggerFactory struct {
+	metadata *trigger.Metadata
+}
+
+//NewFactory create a new Trigger factory
+func NewFactory(md *trigger.Metadata) trigger.Factory {
+	return &TriggerFactory{metadata: md}
+}
+
+//New Creates a new trigger instance for a given id
+func (t *TriggerFactory) New(config *trigger.Config) (trigger.Trigger, error) {
+	settings := &Settings{}
+	err := metadata.MapToStruct(config.Settings, settings, true)
+	if err != nil {
+		return nil, err
+	}
+	if settings.Connection != "" {
+		mConn, toConnerr := coerce.ToConnection(settings.Connection)
+		if toConnerr != nil {
+			return nil, toConnerr
+		}
+		mclient := mConn.(*mongodb.MongodbSharedConfigManager).GetClient()
+		return &Trigger{metadata: t.metadata, settings: settings, id: config.Id, mclient: mclient}, nil
+	}
+	return nil, nil
+}
+
+// Metadata implements trigger.Factory.Metadata
+func (*TriggerFactory) Metadata() *trigger.Metadata {
+	return triggerMd
+}
+
+// Trigger is a stub for your Trigger implementation
+type Trigger struct {
+	metadata  *trigger.Metadata
+	settings  *Settings
+	evntLsnrs []*EventListener
+	mclient   *mongo.Client
+	logger    log.Logger
+	id        string
+}
+
+// EventListener is structure of a single EventListener
+type EventListener struct {
+	handler      trigger.Handler
+	collName     string
+	listenInsert bool
+	listenUpdate bool
+	listenRemove bool
+	mclient      *mongo.Client
```

Comment:
Why the duplicate mClient field in this one and Trigger? Might need to come up with better way.

Reply:
Yes, will refactor to get rid of mclient from EventListener struct and just re-use mclient from trigger struct